### PR TITLE
Improve performance of events api

### PR DESCRIPTION
**Pull request by mfenner**
This pull request makes changes described in #503, improving the performance of aggregations and facets in the events API endpoint.

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -139,16 +139,12 @@ def index
 
       render json: EventSerializer.new(results, options).serialized_json, status: :ok
     else
-      aggregations = params.fetch(:aggregations, "") || ""
```
Comment (kjgarza) [inline app/controllers/events_controller.rb:142]: is it truly necessary to stop supporting the `aggregation selection` functionality? This functionality doesn't have an impact in performance.

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -139,16 +139,12 @@ def index
...
-
-      sources = total.positive? && aggregations.blank? || aggregations.include?("query_aggregations") ? facet_by_source(response.response.aggregations.sources.buckets) : nil
-      prefixes = total.positive? && aggregations.blank? || aggregations.include?("query_aggregations") ? facet_by_source(response.response.aggregations.prefixes.buckets) : nil
-      citation_types = total.positive? && aggregations.blank? || aggregations.include?("query_aggregations") ? facet_by_citation_type(response.response.aggregations.citation_types.buckets) : nil
-      relation_types = total.positive? && aggregations.blank? || aggregations.include?("query_aggregations") ? facet_by_relation_type(response.response.aggregations.relation_types.buckets) : nil
-      registrants = total.positive? && aggregations.blank? || aggregations.include?("query_aggregations")  ? facet_by_registrants(response.response.aggregations.registrants.buckets) : nil
-      pairings = total.positive? && aggregations.blank? || aggregations.include?("query_aggregations") ? facet_by_pairings(response.response.aggregations.pairings.buckets) : nil
```
Comment (kjgarza) [inline app/controllers/events_controller.rb:149]: I think the main performance improvement on this PR is about removing `states` and `pairings` from the aggregations. And that can be easy achieved by just removing them out of the "query_aggregations" hash.

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (mfenner) [reply app/controllers/events_controller.rb:142]: The overall code for Elasticsearch aggregations is overly complex and I am not sure what value it adds. Right now the top priority is performance.

File: app/controllers/events_controller.rb (hunk 2 above)
Comment (mfenner) [reply app/controllers/events_controller.rb:149]: The performance of the events API is very poor, and we will not get away with minor changes. This is just the first step.

**Comment by sarala** [comment]
I don't understand all of it but I can see bucket sizes being reduced etc and works locally.

**Pull request merged**